docs/tools.md

```diff
@@ -0,0 +1,117 @@
+
+  <!-- eslint-skip -->
+
+  ```typescript
+  inputSchema() {
+      // any Zod definition is good here, this is typical simple example
+      return z.object({
+        // list of key-value pairs
+      });
+  }
+  ```
+
+- Implement initialisation:
+
+  The unnamed static block is executed when your tool is called for the first time. It is used for registering your tool as `serializable` to the agent and any other custom initialisation that may be required.
+
+  <!-- eslint-skip -->
+
+  ```typescript
+  static {
+      this.register();
+  }
+  ```
+
+- Implement the `_run()` method:
+
+  <!-- eslint-skip -->
+
+  ```typescript
+  protected async _run(input: ToolInput<this>, options?: BaseToolRunOptions) {
+      // insert custom code here
+      // MUST: return an instance of the output type specified in the tool class definition
+      // MAY: throw an instance of ToolError upon unrecoverable error conditions encountered by the tool
+  }
+  ```
+
+### Using tools with agents
+
+In order for a tool to be of some utility within an agent, you must enable the agent with knowledge of the tool. To do this, the tool code module must be imported into the agent and passed to the tools array during creation of a `BeeAgent`. An example can be found in the [bee agent](../examples/agents/bee.ts) or you can use a code snippet such as the one below that creates an agent with the built-in [ArXiv tool](../src/tools/arxiv.ts):
+
+```typescript
+import { ArXivTool } from "bee-agent-framework/tools/arxiv";
+
+const llm = new OllamaChatLLM({
+  modelId: "insert-model-id-here",
+});
+
+const agent = new BeeAgent({
+  llm,
+  memory: new TokenMemory({ llm }),
+  tools: [new ArXivTool()],
+});
+```
+
+## Examples
+
+### Hello World
+
+The simplest form of an example tool is the [helloworld](../examples/tools/helloworld.ts) tool. This example implements the Tool class and will simply prepend the word "Hello" to a given input such as "Please give a special greeting to Bee!". It is not intended for usage with any agents since the functionality provided is highly trivial. However, it may serve as a starter template for writing new tools.
+
+### Open Library
+
+The [openlibrary](../examples/tools/helloworld.ts) tool allows an agent to query the [Open Library](https://openlibrary.org/) via its [book search API](https://openlibrary.org/dev/docs/api/search). This functionality injects knowledge about book metadata (not book content) into an agent. It serves as an example for several key aspects of tool creation:
```

Comment:
invalid path

Reply:
Resolved (pending further review) in d72dc64